arrivals/src/main/scala/com/pagerduty/arrivals/impl/authproxy/AuthProxyController.scala

```diff
@@ -143,14 +145,19 @@ trait AuthProxyController[AuthConfig <: HeaderAuthConfig, AddressingConfig] {
       responseFilter: ResponseFilter[Option[AuthConfig#AuthData]]
     )(implicit reqMeta: RequestMetadata
     ): Route = {
-    complete {
-      headerAuthenticator.addAuthHeader(authConfig)(
-        request,
-        requiredPermission.asInstanceOf[Option[authConfig.Permission]]
-      ) {
-        case (authedRequest, optAuthData) =>
-          authProxyRequestHandler.apply(authedRequest, upstream, optAuthData, httpProxy, requestFilter, responseFilter)
-      }
+    authenticate(authConfig)(requiredPermission.asInstanceOf[Option[authConfig.Permission]])(reqMeta, metrics) {
+      optAuthData =>
+        filterRequest(requestFilter, optAuthData) {
+          addAuthHeader(authConfig)(optAuthData.asInstanceOf[Option[authConfig.AuthData]])(reqMeta) {
```

Comment:
can we pull `addAuthHeader(authConfig)` out? We shouldn't have to reapply it for every request.
ie. `authHeaderAdder = addAuthHeader(authConfig)`
and replace this line with `authHeaderAdder(...)`

Reply:
Interesting idea, but it doesn't compile and I'm not quite sure why. Seems to be implicit related, but could also be related to the use of a path-dependent type?